src/main/java/de/dennisguse/opentracks/publicapi/StartRecording.java

```diff
@@ -1,11 +1,23 @@
 package de.dennisguse.opentracks.publicapi;
 
+import android.content.Intent;
+import android.os.Bundle;
+
+import de.dennisguse.opentracks.data.models.Track;
 import de.dennisguse.opentracks.services.TrackRecordingService;
+import de.dennisguse.opentracks.util.IntentDashboardUtils;
 
 public class StartRecording extends AbstractAPIActivity {
 
     protected void execute(TrackRecordingService service) {
-        service.startNewTrack();
+        Track.Id trackId = service.startNewTrack();
+        Intent intent = getIntent();
+        Bundle bundle = intent.getExtras();
+        if (bundle != null) {
+            String originatingPackage = bundle.getString("ORIGINATING_PACKAGE", null);
```

Comment:
Could we do something like this?
https://stackoverflow.com/a/6783661

Reply:
Hmm, `ORIGINATING_PACKAGE` can be freely set by the calling application, right?
What about an API key generated/shown in the OpenTracks settings, which needs to be copied into the calling application, which then needs to provide as an bundle string?

Reply:
I'm afraid in my use case using `startActivityForResult()` isn't an option. It's only available in `Activity`, while the context in Gadgetbridge that's starting this intent is a `Service`.

Reply:
BTW I like the API key suggestion, but I'm holding off on implementing it until after further reviews/comments.